localstack-core/localstack/services/apigateway/resource_providers/aws_apigateway_method.py

```diff
@@ -100,75 +99,45 @@ def create(
         if integration:
-            api_id = model["RestApiId"]
-            res_id = model["ResourceId"]
-
-            kwargs = keys_to_lower(deepcopy(integration))
-            if uri := integration.get("Uri"):
-                kwargs["uri"] = uri
-
-            integration_responses = kwargs.pop("integrationResponses", [])
-            method = model.get("HttpMethod")
-
             apigw.put_integration(
-                restApiId=api_id,
-                resourceId=res_id,
-                httpMethod=method,
-                **kwargs,
+                restApiId=model.get("RestApiId"),
+                resourceId=model.get("ResourceId"),
+                httpMethod=model.get("HttpMethod"),
+                **util.convert_request_kwargs(
+                    integration, operation_model("PutIntegration").input_shape
+                ),
             )
 
+            integration_responses = integration.pop("IntegrationResponses", [])
             for integration_response in integration_responses:
-                integration_response["statusCode"] = str(integration_response["statusCode"])
-                if selection_pattern := integration_response.get("selectionPattern"):
-                    integration_response["selectionPattern"] = str(selection_pattern)
-
                 apigw.put_integration_response(
-                    restApiId=api_id,
-                    resourceId=res_id,
-                    httpMethod=method,
-                    **keys_to_lower(integration_response),
+                    restApiId=model.get("RestApiId"),
+                    resourceId=model.get("ResourceId"),
+                    httpMethod=model.get("HttpMethod"),
+                    **util.convert_request_kwargs(
+                        integration_response, operation_model("PutIntegrationResponse").input_shape
+                    ),
                 )
 
-        responses = model.get("MethodResponses") or []
+        responses = model.get("MethodResponses", [])
         for response in responses:
-            api_id = model["RestApiId"]
-            res_id = model["ResourceId"]
-            kwargs = {}
-            if (response_parameters := response.get("ResponseParameters")) is not None:
-                kwargs["responseParameters"] = response_parameters
-            if (response_models := response.get("ResponseModels")) is not None:
-                kwargs["responseModels"] = response_models
+            kwargs = util.convert_request_kwargs(
+                response, operation_model("PutMethodResponse").input_shape
+            )
 
             apigw.put_method_response(
-                restApiId=api_id,
-                resourceId=res_id,
-                httpMethod=model["HttpMethod"],
-                statusCode=str(response["StatusCode"]),
+                restApiId=model.get("RestApiId"),
+                resourceId=model.get("ResourceId"),
+                httpMethod=model.get("HttpMethod"),
                 **kwargs,
```

Comment:
nit: Some of the uses of the utils are done inline and some first create a `kwargs` variable. It doesn't seem like `kwargs` is used otherwise. Might be more consistent to always use inline? Unless I am missing something? :thinking: